internal/util/validation/validation.go

```diff
@@ -107,6 +113,68 @@ func ValidateBearerToken(token *string, path string) []error {
 	return asErrors(errs)
 }
 
+func ValidateCSRUsages(u *[]string) []error {
+	errs := field.ErrorList{}
+	requiredAllOf := []string{"client auth", "CA:false"}
+	notAllowed := []string{}
+
+	for _, usage := range *u {
+		for _, na := range notAllowed {
+			if usage == na {
+				err := fmt.Sprintf("usage not allowed: %s\n", na)
+				errs = append(errs, field.Invalid(fieldPathFor("spec.usages"), u, err))
+			}
+		}
+		for i, r := range requiredAllOf {
+			if usage == r {
+				// remove fulfilled req from list
+				requiredAllOf = append(requiredAllOf[:i], requiredAllOf[i+1:]...)
+			}
+		}
+	}
+	if len(requiredAllOf) > 0 {
+		err := fmt.Sprintf("required usages must be present in request: %s\n", requiredAllOf)
+		errs = append(errs, field.Invalid(fieldPathFor("spec.usages"), u, err))
+	}
+	return asErrors(errs)
+}
+
+// NOTE: ValidateSignerName will not error and is a placeholder for when FC has multiple signers.
+// Currently every request is sent to the only signer, named "ca" and defined in cmd/flightctl-api/main.go
+func ValidateSignerName(s string) []error {
+	//errs := field.ErrorList{}
```

Comment:
Why not verify that the only signer is "ca"?
If it needs to be commented out, why not remove the code?

Reply:
It's mainly a placeholder for the future and we do not want to cause an error on this for the user currently, but I take your point. I added this back in so it will verify the signer (see #453 where these commits belong), which is a value that can just be generated internally for the moment if we want.